src/mid.cpp

```diff
@@ -943,19 +970,33 @@ void CmidPlayer::rewind(int subsong)
             case FILE_MIDI:
                 if (type != FILE_LUCAS)
                     tins=128;
-                getnext(11);  /*skip header*/
+                getnext(11); /* skip past header data until deltas is reached */
                 deltas=getnext(2);
                 midiprintf ("deltas:%ld\n",deltas);
-                getnext(4);
 
                 curtrack=0;
-                track[curtrack].on=1;
-                track[curtrack].tend=getnext(4);
-                midiprintf ("tracklen:%lu\n",track[curtrack].tend);
-                //track[curtrack].tend += pos; // FIXME: length -> end position
-                if (track[curtrack].tend > flen) // no music after end of file
-                    track[curtrack].tend = flen;
-                track[curtrack].spos=pos;
+                while ((curtrack == 0) ||
+                       ((midi_type == 1) && (curtrack < 16)))
+                {
+                    /* MIDI type 0 (and LucasArts AdLib MIDI) stores all the MIDI channels in a single track,
+                     * while MIDI type 1 splits each channel into separate tracks */
```

Comment:
Well actually MIDI type-1 may have more than 16 tracks, and some of them may share one channel, like drum tracks (one track per drum type, but they all use channel 10 anyway).